topic_tools/src/demux_node.cpp

```diff
@@ -0,0 +1,186 @@
+// Copyright 2024 Rufus Wong
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "topic_tools/demux_node.hpp"
+
+#include <memory>
+#include <optional>  // NOLINT : https://github.com/ament/ament_lint/pull/324
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "rclcpp/rclcpp.hpp"
+
+namespace topic_tools
+{
+DemuxNode::DemuxNode(const rclcpp::NodeOptions & options)
+: ToolBaseNode("demux", options)
+{
+  using std::placeholders::_1;
+  using std::placeholders::_2;
+
+  output_topic_ = declare_parameter("initial_topic", "");
+  input_topic_ = declare_parameter("input_topic", "~/input");
+  lazy_ = false;
+  output_topics_ = declare_parameter<std::vector<std::string>>("output_topics");
+  if (output_topic_.empty()) {
+    output_topic_ = output_topics_.front();
+  }
+
+  discovery_timer_ = this->create_wall_timer(
+    discovery_period_, std::bind(&DemuxNode::make_subscribe_unsubscribe_decisions, this));
```

Comment:
What value is assigned to the `discovery_period_`?
 I wasn't able to find where it is declared or assigned.

Reply:
It's assigned in the base class [`ToolBaseNode`](https://github.com/rcywongaa/topic_tools/blob/add_demux/topic_tools/include/topic_tools/tool_base_node.hpp#L43), currently it's `std::chrono::milliseconds{100};`